src/Lean/Parser/Command.lean

```diff
@@ -754,6 +754,30 @@ list, so it should be brief.
   optional (docComment >> ppLine) >>
   "tactic_extension " >> ident
 
+/--
+Document a recommended spelling for a notation in identifiers.
+
+```
+/-- some additional info -/
+recommended_spelling "∧" "and" And «term_∧_»
+```
+
+will do the following:
+* Adds the sentence "The recommended spelling of `∧` in identifier is `and` (some additional info)."
+  to the end of the docstring for `And` and for `∧`. If the additional info is more than a single
+  line, it will be placed below the sentence instead of in parentheses.
+* Registers this information in an environment extension, so that it will later be possible to
+  generate a table with all recommended spellings.
+
+You can add attach the recommended spelling to as many declarations as you want. It is recommended
+to attach the recommended spelling to all relevant parsers as well as the declaration the parsers
+refer to (if such a declaration exists). Note that the `inherit_doc` attribute does *not* copy
+recommended spellings, so even though the parser for `∧` uses `@[inherit_doc And]`, we have to
```

Comment:
This isn't really an actionable comment for this PR, but this context is useful for it. It seems that we're working at too low of an abstraction here - a great many notations are fundamentally alternative ways of writing applications of some predetermined name (whether it be monomorphic like `And` or overloaded like `HAdd.hAdd` or something a bit more complicated like `Seq.seq` or `ite`/`dite`). That seems like a fact that we should be able to tell Lean about, rather than thinking about "copy the docs, copy the recommended spellings, etc". That's what #4490 did for tactics.

Reply:
I agree that this would be nice to have. One way to approach this might be to have an attribute that you can attach to term parsers that allows you to tell Lean that a certain term parser "belongs" to a certain constant (like `And`). But it seems to me that this can be done independently of this PR, and once we have this `notation_for` attribute, we can get rid of the `inherit_doc` attributes and the parser names in `recommended_spelling`s in one go after that.

Reply:
I agree fully.